hacspec/src/lib.rs

```diff
@@ -755,31 +760,36 @@ fn parse_suites_i(
         // the CBOR array length is encoded in the first byte, so we extract it
         let suites_len: U8 = rcvd_message_1.content[1] - U8(CBOR_MAJOR_ARRAY);
         let suites_len: usize = suites_len.declassify().into();
-        raw_suites_len = 1; // account for the CBOR_MAJOR_ARRAY byte
-        if suites_len <= EDHOC_SUITES.len() {
-            let mut error_occurred = false;
-            for j in 0..suites_len {
-                raw_suites_len += 1;
-                if !error_occurred {
-                    // parse based on cipher suite identifier
-                    let cs_id = (rcvd_message_1.content[raw_suites_len] as U8).declassify();
-                    if cs_id >= 0x00 && cs_id <= 0x17 {
-                        // CBOR unsigned integer (0..23)
-                        suites_i[j] = rcvd_message_1.content[raw_suites_len];
-                        suites_i_len += 1;
-                    } else if cs_id == 0x18 {
-                        // CBOR unsigned integer (one-byte uint8_t follows)
-                        raw_suites_len += 1; // account for the 0x18 tag byte
-                        suites_i[j] = rcvd_message_1.content[raw_suites_len];
-                        suites_i_len += 1;
-                    } else {
-                        error = EDHOCError::ParsingError;
-                        error_occurred = true;
+        // check surplus array encoding of ciphersuite
+        if suites_len > 1 {
```

Comment:
Any particular reason why the checks on lines 764 and 766 are not grouped?

Reply:
What does "surplus" array encoding mean?

Reply:
It means that the encoding as array uses extra bytes, since it could have been only an integer.